client/serialize/rust/src/deserializer/old.rs

```diff
@@ -0,0 +1,234 @@
+use bimap::BiHashMap;
```

Comment:
This file was moved from the module and renamed to `old`. I reused some of this code in the second attempt at implementing the deserializer. So you can ignore this file - just review the stuff in the `new` module.